src/utils.c

```diff
@@ -62,7 +62,7 @@ read_until_eof(FILE *stream) {
     size_t n = fread(buf, 1, 1024, stream);
     len += strlen(buf);
     str = realloc(str, len);
-    strcat(str, buf);
+    strncat(str, buf, n);
```

Comment:
`len` represents the total length of read so far , and  `n` represents the length of each time read. so we only need to concatenate `n`  items to  `str`.